### ISSUE TITLE
  Unknown CMake command "catch_discover_tests".

### ISSUE THREAD

**Issue by xgshark** [wolo-research/bulldog#3]
I am trying to build this on linux.

When I follow the build instructions I get this error upon cmake..

> 

CMake Error at modules/core/CMakeLists.txt:49 (include):
  include could not find load file:

Catch

CMake Error at modules/core/CMakeLists.txt:52 (catch_discover_tests):
  Unknown CMake command "catch_discover_tests".

<

I tried to install Catch manually and using git submodule without success. 

Sorry if this is basic but how can I proceed from here?

**Comment by kmurf1999** [comment]
Make sure you're using the right branch of catch in your 3rdparty folder.  Check the .gitmodules file for the correct branch

**Comment by dpmaloney** [comment]
@kmurf1999 the branch of Catch2 in that file isn't even correct so I'm not sure how you got it to build. I believe you need the v2.x branch.

**Comment by bluedevils23** [comment]
there is no master branch, switching to v2.x solved the issue.

**Comment by charliertm** [comment]
@bluedevils23 were you able to successfully make after building with v2.x ? didn't work for me.

**Comment by dpmaloney** [comment]
Yeah, no clue what version of spdlog they used but it seems like it doesn't do formatting of file pointers anymore so you'll either need to figure out how to find the version they used, or just remove or reformat all the places that they log a file pointer. The make process should throw some errors that tell you exactly where, but you'll still need to fix a problem with the way they do multithreading as well as some other segfaults that happen with the tests. I'll probably try to update my fork with these fixes soon.

**Comment by bluedevils23** [comment]
@dpmaloney v.1.5.0 works

**Comment by xgshark** [comment]
Does anybody have a step by step guide on how to build this on linux?

**Comment by charliertm** [comment]
@xgshark i think most of the steps you should need to get it to build are here but drop your discord and I can walk you through how I got it to build if you want.

**Comment by bluedevils23** [comment]
> @bluedevils23 were you able to successfully make after building with v2.x ? didn't work for me.

Yes v2.x works for me, paste error log here?

**Comment by parmercapital** [comment]
@charliertm I'm also trying to set this up and running into issues, would appreciate the walkthrough when you have a chance. Discord: ParmerJohn#4893. Thanks.
